### PR TITLE
Add a command to retrieve the state of a mapping

### PR DESCRIPTION
**Pull request by gulopine**

### CODE REVIEW

File: lib/commands/connect/mapping-state.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  command: 'mapping:state',
+  description: 'return a mapping state',
+  help: 'return a mapping state',
+  args: [ {name: 'mapping'}],
+  flags: [
+    {name: 'resource', char: 'r', description: 'specific connection resource name', hasValue: true}
+  ],
+  needsApp: true,
+  needsAuth: true,
+  run: cli.command(co.wrap(function* (context, heroku) {
```
Comment (iamjem) [inline lib/commands/connect/mapping-state.js]: Can you safely use generators in these plugins? CO states in their documentation that you have to run node with harmony flags for any of this to work.

**Comment by iamjem** [comment]
Same comment as other PR, sssuming the generator stuff is safe to use, this looks :+1: 

I must say, this helps make node's async spaghetti more synchronous-looking. I'll have to read more about how errors are handled with the native spec, co's error handling looks simple enough.

File: lib/commands/connect/mapping-state.js (hunk 1 above)
Comment (gulopine) [inline lib/commands/connect/mapping-state.js]: co and generators are recommended in [Heroku's toolbelt plugin development docs](https://devcenter.heroku.com/articles/developing-toolbelt-plug-ins#using-generators-with-co), and they've worked fine for me so far.

**Comment by gulopine** [comment]
The nice thing about error handling between `co` and `cli.command` is that any time we reject a promise and don't explicitly do anything else with it, it'll automatically get nicely formatted output that's consistent with the rest of the toolbelt. So they get the little red arrow and an indent before the actual message.

There are some cases where we need to handle errors specifically, and so far, I've find that easier to do that with the traditional pair of `.then` and `.catch`, but I'm probably just not familiar enough with `co` error handling to do it "better".

That said, I haven't managed to get generators to work inside of a promise for some reason, so there are other cases where I'm still using `.then` when I'd otherwise use a `yield`. Roughly, this is all a first draft, and I'm sure we can improve it over time. It's already a big win, though. :grinning:

**Pull request merged**